src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -296,14 +472,58 @@ public void SetOptions(DacDeployOptions options)
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-
+                // Note that we set excluded object types here since dacfx has this value as null;
                 if (prop != null)
                 {
-                    deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
                 }
             }
         }
 
+        /// <summary>
+        /// Sets the Value and Description to all properties
+        /// </summary>
+        /// <param name="prop"></param>
+        /// <param name="val"></param>
+        /// <param name="deployOptionsProp"></param>
+        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        {
+            var val = prop.GetValue(options);
+            var attribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault();
+
+            if (prop.PropertyType == typeof(ObjectType[]))
```

Comment:
My PR solves needing to have these if/else statements which greatly reduces the duplication and also has the benefit of not hardcoding the "supported" types so adding a new type later on is easier. I'd highly recommend looking at taking in my solution there instead of reverting back to this...

Reply:
Oh No! I missed reviewing the total changes of your PR, I barely saw few comments and thought those are all changes! let me work on the suggessions! Thanks

Reply:
And note that the purpose of that PR was that you could merge it into your branch and thus not require actually making the changes yourself. So if things look good just click "squash and merge" and then go pull down the latest from your branch.

Reply:
Actually, I have copied the changes and made all updates and later saw the comment about merging. since already added these changes can we close the other PR or is there a way to merge that into it. need help here! Thanks :)